low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -98,21 +98,26 @@ class JoinClusterCmd(Cmd):
 
     def __init__(self, args):
         super().__init__(args)
+        self.nodes = None
+
+    @staticmethod
+    def add_args(parser: str, cls: str, name: str):
+        """Add Command args for parsing."""
+        parsers = parser.add_parser(cls.name, help='%s' % cls.__doc__)
+        parsers.add_argument('args', nargs='*', default=[], help='args')
+        parsers.add_argument('--nodes', nargs='*', default=[], help='Node names separted by comma')
+        parsers.set_defaults(command=cls)
 
     def validate(self):
-        if not self.args:
+        if not self.args.nodes:
             raise SetupError(1,
                              "Validation failure. %s",
                              "join_cluster requires comma separated node names as argument.")
-        if (len(self.args) != 2) or (self.args[0] != "--nodes"):
-            raise SetupError(1,
-                             "%s - Argument validation failure. %s",
-                             self.name,
-                             "Check usage.")
+        self.nodes = self.args.nodes[0]
 
     def process(self):
         from cortx.sspl.bin.setup_rabbitmq_cluster import RMQClusterConfiguration
-        RMQClusterConfiguration(self.args[1]).process()
+        RMQClusterConfiguration(self.nodes).process()
```

Comment:
JoinClusterCmd invoked by dev deploy only right.
After message bus integration, rmq node names wont be needed/passed & same code here should work, so possibly process() should be invoked only if nodes are passed?

Reply:
`JoinClusterCmd invoked by dev deploy only right.` Yes
`so possibly process() should be invoked only if nodes are passed?` I will add a check here.

Reply:
Added a condition under process().